R/derive_date_vars.R

```diff
@@ -6,22 +6,38 @@
 #' @param dtc The `'--DTC'` date to impute
 #'
 #'   A character date is expected in a format like `yyyy-mm-dd` or
-#'   `yyyy-mm-ddThh:mm:ss`. If the year part is not recorded (missing date), no
-#'   imputation is performed.
+#'   `yyyy-mm-ddThh:mm:ss`. Trailing components can be omitted and `-` is a
+#'   valid value for any component.
 #'
-#' @param date_imputation The value to impute the day/month when a datepart is
+#' @param highest_imputation Highest imputation level
+#'
+#'   If a component at a higher level than the highest imputation level is
+#'   missing, `NA_character_` is returned. For example, for `highest_imputation
+#'   = "D"` `"2020"` results in `NA_character_` because the month is missing.
```

Comment:
```suggestion
#'   If a component at a higher level than the highest imputation level is
#'   missing, `NA_character_` is returned. For example, a partial date "2020" will have for  the 
#'  `highest_imputation' = "D"` and results in `NA_character_` because the month is missing.
```
This examples seems slightly  backwards, but I might not be understanding it.

Reply:
I have added another paragraph. I hope it makes it clearer.

I am not sure about your suggested change. To be honest, I do not understand it.

Reply:
Sorry I think I was having trouble understanding the language here.  Apologies for confusion